File: demo/src/main/java/com/midas/demo/mvp/model/HotTabModel.java
package com.midas.demo.mvp.model;

import com.midas.demo.mvp.model.bean.TabInfoBean;
import com.midas.demo.net.RetrofitManger;
import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class HotTabModel {

    /**
     * 获取 TabInfo
     */
    public Observable<TabInfoBean> getTabInfo() {
        return RetrofitManger.getApiService().getRankList()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread());
    }

}

File: app/src/main/java/com/midas/eye/mvp/model/HotTabModel.kt
package com.midas.eye.mvp.model

import com.midas.eye.mvp.model.bean.TabInfoBean
import com.midas.eye.net.RetrofitManager
import com.midas.eye.rx.scheduler.SchedulerUtils
import io.reactivex.Observable

/**
 * Created by midas on 2019/05/28.
 * desc: 热门 Model
 */
class HotTabModel {

    /**
     * 获取 TabInfo
     */
    fun getTabInfo(): Observable<TabInfoBean> {
        return RetrofitManager.service.getRankList()
                .compose(SchedulerUtils.ioToMain())
    }

}
